File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

// get key
int main(int argc, string argv[])
{
    if (argc == 2)
    {
        //convert argv to integer
        string num = argv[1];
        int k = atoi(num);
        //printf("%i\n", k);
        
        //get plain text
        printf("plaintext: ");
        string p = get_string();
        //printf("%s\n", p);
        
        //encipher plain text
        //iterate over the characters in plain text one character at a time
        
         printf("ciphertext: ");
         
        for (int i = 0, n = strlen(p); i < n; i++) {
            
            //if alphabetic apply key
            if (isalpha(p[i])) 
            
            {
                if (isupper(p[i]))
                {
                    // c = (p + k) % 26 - convert to alpha index
                    int a = p[i] - 65;
                    int c = (a + k)%26;
                    //printf("%i ", c);
                    printf("%c", 'A' + c);
                    //printf(" - %c\n", (p[i]));
                }
                else if (islower(p[i]))
                {
                    int a = p[i] - 97;
                    int c = (a + k)%26;
                    printf("%c", 'a' + c);
                }
            }
            else
            
            // else print number or symbol as is
            {
                printf("%c", p[i]);
            }
            
        }
        printf("\n");
    }
    else
    {
        printf("Usage: %s\n", argv[0]);
        return 1;
    }
    return 0;

}

File: pset6/caesar.py
import cs50
import sys

if len(sys.argv) == 2:
    num = sys.argv[1]
    
    #convert argv to integer
    k = int(num)
   
    #get plaintext
    p = input("plaintext: ")
    
    #encipher plain text
    #iterate over each char in plain text
    print("ciphertext: ",end="")
    
    for char in p:
        if char.isalpha():
            
            if char.isupper():
                a = ord(char) - 65
                c = ((a+k)%26) + 65
                print(chr(c),end="")
            
            elif char.islower():
                a = ord(char) - 97
                c = ((a+k)%26) + 97
                print(chr(c),end="")
                
        else:
            print(char,end="")
    
    print()
    
else:
    print("Usage: "+ sys.argv[0])
    
